apps/trackers/jira/query.py

```diff
@@ -267,3 +271,89 @@ def generate_additional_fields(self):
                     continue
 
                 self._query["fields"][PS_ADDITIONAL_FIELD_TO_JIRA[name]] = field_value
+
+    def generate_cc(self):
+        """
+        generate query for CC list
+        """
+
+        # Each instance of TrackerJiraQueryBuilder is used only once, but if ever used twice,
+        # always produce consistent query and comment.
+        self._comment = None
+
+        if self.tracker.external_system_id:
+            # Add CCs only on creation.
+            return
+
+        # NOTE That SFM2 for Jira tracker creation uses ps_module.component_overrides only for
+        #      generating Jira "components" field, which OSIDB doesn't do, but not for CC lists;
+        #      CC list creation is based solely on ps_component, not on bz_component.
+        #      Therefore AffectCCBuilder.ps2bz_component is not reused here.
+        #      TODO: Is this a bug?
+        # Parse BZ component
+        if self.ps_component and "/" in self.ps_component:
+            bz_component = self.ps_component.split("/")[-1]
+        else:
+            bz_component = self.ps_component
+
+        # embargoed value unused here
+        affect_cc_builder = AffectCCBuilder(
+            self.tracker.affects.first(), embargoed=None
+        )
+        # TODO: Why does AffectCCBuilder set bz_component to None?
+        affect_cc_builder.bz_component = bz_component
+        cc_list = set(affect_cc_builder.component_cc())
+
+        if self.ps_module.default_cc:
+            # Default CC List
+            cc_list.update(self.ps_module.default_cc)
+        if self.tracker.is_embargoed and self.ps_module.private_tracker_cc:
+            cc_list.update(self.ps_module.private_tracker_cc)
```

Comment:
I see that there is no `private_tracker_cc` in `AffectCCBuilder.module_cc`. That may be a bug. Could you rather improve that method and use it here instead of implementing it again?

Reply:
Yes, will fix accordingly.

Confirming that SFM2 does generate affect CC lists from private_tracker_cc, so adding this in OSIDB as you suggest will fix an OSIDB bug and bring it in line with SFM2 (the class is called AffectCCBuilder after all).

- cc_list_for_affects method calls cc_list_from_prod_defs method which generates ccs from private_tracker_cc
- cc_list_for_affects method is called by cc_list_for_flaw method which is called from FlawBase class so its's used in SFM2

Also confirming that AffectCCBuilder is used to build affect CC lists in OSIDB already (given the [confusion](https://redhat-internal.slack.com/archives/C061QHHTM2M/p1712342079811019) I wasn't sure).

Reply:
Fixed in the revised version.